### DISCUSSION TITLE
Support for privately-hosted ruby gems

### DISCUSSION THREAD

**Discussion by elondres-mim** [renovatebot/renovate#29576] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab via the CI template

### Please tell us more about your question or problem

My application has several dependencies which are hosted as private git repositories. Renovate seems unable to create MRs for any dependency in my project, including the many public gem dependencies even though it can post to the dependency dashboard with the exact version it would like to update to, because it's running `bundler lock --patch --strict --update` which tries to rebuild the entire lockfile. As I'm using the CI template I'm not sure there's an easy way to provide renovate with configuration to pull internal git repositories from within the same gitlab instance. Has anyone successfully used renovate like this before?

### Logs (if relevant)

<details><summary>Logs</summary>

```
Host key verification failed.\r\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists.\nGit error: command `git clone --bare --no-hardlinks --quiet --no-tags --depth 1\n--single-branch --\nssh://**redacted**@gitlab.instance.hostname/path/to/gem/repository.git\
Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Users have definitely used Bundler with private gems before. Do the lookups for the private gems succeed too, or only for public gems? Look for them in the "packageFiles with updates" log message.

**Reply by elondres-mim** [reply]
The lookups for the private gems were failing, which makes sense to me because the template renovate job doesn't have access to them.

 I tried to add them to ignored deps, but then with other MRs for public gems I get the lockfile error I mentioned in the OP.

**Reply by rarkins** [reply]
Is there a bundler command which could:
- Update the lock file for public gems, but
- Ignore private gems?

I'm not sure if this is really possible. It would be better if you get authentication working for the private gems.

**Reply by elondres-mim** [reply]
Do you have any thoughts on how to get authentication working? Right now I am just using the provided CI template, I am not sure how to add an SSH key and key configuration to it.

**Reply by rarkins** [reply]
Do you support any https authentication methods or only SSH? SSH could work for bundler (just set up the key independent of Renovate) but Renovate supports http-based lookups only.

**Reply by elondres-mim** [reply]
Right now we only support SSH, I'm not sure how we would support HTTP without lots of additional work, but if Renovate doesn't support authenticating via SSH, then that's ok :) Thank you

**Reply by viceice** [reply]
your best option is to configure ssh agent with loaded key before running renovate and then allow passing the ssh agent env vars to bundler.

see allowedEnv in docs
